cpp/include/functions.h

```diff
@@ -219,22 +203,6 @@ void get_two_hop_neighbors(Graph* graph, gdf_column* first, gdf_column* second);
 /* ----------------------------------------------------------------------------*/
 void snmg_csrmv (size_t * part_offsets, gdf_column * off, gdf_column * ind, gdf_column * val, gdf_column ** x_col);
 
-/**
- * @Synopsis   Computes degree(in, out, in+out) of all the nodes of a Graph
- *
- * @Param[in]* graph                 in  : graph descriptor with graph->transposedAdjList or graph->adjList present
- * @Param[in] x                      in  : integer value indicating type of degree calculation
- *                                         0 : in+out degree
- *                                         1 : in-degree
- *                                         2 : out-degree
- *
- * @Param[out] *degree               out : gdf_column of size V (V is number of vertices) initialized to zeros.
- *                                         Contains the computed degree of every vertex.
- *
- * @throws     cugraph::logic_error when an error occurs.
- */
-/* ----------------------------------------------------------------------------*/
-void degree(Graph* graph, gdf_column *degree, int x);
```

Comment:
NOTE: I made degree a function on the Graph object.  This seemed more like a function of the graph to me.  It also allowed me to easily customize the implementation based on whether the graph was CSR/CSC (where the computation is just a subtraction on the offsets array) or COO (where the computation is a histogram)

Reply:
what about degrees (get in and out degree rather than just total count)?

Reply:
degrees is implemented in python as two calls to this function, with x = 1 and x = 2.  That's how it was, I didn't think about changing it.  The only real performance impact would be that the vertex id column gets created an extra time in the degrees call.